### PR TITLE
Added "total views" in the my channels list

### PR DESCRIPTION
**Pull request by Poslovitch**
## Description

I added a "total views" counter for each channel in the "my-channels" list. (1)

Also made the followersCount consistent with other counters (if =0, "No followers"). (2)

However, I couldn't find a way to get around the "pagination" in the VideoService.

## Related issues

Closes https://github.com/Chocobozzz/PeerTube/issues/4331

## Has this been tested?

- [x] 🙅 no, because this PR does not update server code

## Screenshots

image

### CODE REVIEW

**Comment by Poslovitch** [comment]
Alright, now that I've added a "column" to VideoChannel, I guess I should add a migration in server/initializers that would add the column in that database table. Am I right?

**Comment by Chocobozzz** [comment]
> Alright, now that I've added a "column" to VideoChannel, I guess I should add a migration in server/initializers that would add the column in that database table. Am I right?

No, prefer calculating the view *on the fly* instead (like we do for `viewsPerDay`)

**Comment by smilekison** [comment]
Is the video view updated in real time? Can you tell me where can i find the database schema for PeerTube?

File: server/models/video/video-channel.ts [hunk 1]
```diff
@@ -287,6 +287,12 @@ export type SummaryOptions = {
               ')'
             ),
             'viewsPerDay'
+          ],
+          [
+            literal(
+              '(SELECT COALESCE(SUM("video".views), 0) AS totalViews FROM "video" WHERE "video"."channelId" = "VideoChannelModel"."id" GROUP BY "video"."channelId")'
```
Comment (Poslovitch) [inline server/models/video/video-channel.ts:293]: I have a lot of trouble with that query.
All I get while running it is:
```
Backend returned code 500, errorMessage is: syntax error at or near "SELECT"
```

And if I run it inside psql, here's what it gives me:
```
ERROR:  missing FROM-clause entry for table "VideoChannelModel"
LINE 1: ...talViews FROM "video" WHERE "video"."channelId" = "VideoChan...
                                                             ^
```

Comment (Chocobozzz) [reply server/models/video/video-channel.ts:293]: Backend seems to work well on my side

Comment (Poslovitch) [reply server/models/video/video-channel.ts:293]: You mean, the query's working?

Comment (Chocobozzz) [reply server/models/video/video-channel.ts:293]: Yes and the controllers correctly set the `totalViews` attribute for channels

Comment (Poslovitch) [reply server/models/video/video-channel.ts:293]: Okay. That may be caused by Gitpod then. I'll investigate that further, some time in the week.

**Comment by Chocobozzz** [comment]
Closing due to inactivity, feel free to comment/push commits if you want to reopen :)

**Comment by Poslovitch** [comment]
Hello there. I can't get that query to work on Gitpod.
So, let's assume *it works*. What else do I have to do to finish the PR? :slightly_smiling_face:

**Comment by Chocobozzz** [comment]
>  What else do I have to do to finish the PR?

Unit tests :)

**Comment by Chocobozzz** [comment]
Closing due to inactivity. Does not hesitate to ask for the reopening :)

**Pull request closed without merging**